lib/http/request.rb

```diff
@@ -95,13 +95,19 @@ def initialize(opts)
     end
 
     # Returns new Request with updated uri
-    def redirect(uri, verb = @verb)
+    def redirect(uri, verb = @verb, orig_uri = nil)
       headers = self.headers.dup
       headers.delete(Headers::HOST)
 
+      uri = @uri.join(uri)
+      if !orig_uri.nil? && (uri.host != orig_uri.host)
+        headers.delete(Headers::AUTHORIZATION)
+        headers.delete(Headers::COOKIE)
```

Comment:
A suggested path forward here:

Extract an `Http::Session` class, and give it a first sense of an authorization token (e.g. `auth_token`?) and cookies, then selectively decide whether or not to *add* those to the headers of particular origins. That way it avoids the need to munge these headers.

This should also make it relatively straightforward to address #511 as well.

Reply:
IMO HTTP::Session should get it's behavior inspiration from user-oriented web clients (AKA Firefox and alike). And those are simply dropping ALL the headers upon any redirect (from what I observed), but naturally add Cookie header (based on host + location) of request hop.